### ISSUE TITLE
Enable SSO Across Applications

### ISSUE THREAD

**Issue by tomjackman** [feedhenry/mobile-security-ios-template#17]
## Reference the AppAuth-iOS Master Branch which has Custom Browser Support in the Podfile.
`pod 'AppAuth', :git => 'https://github.com/openid/AppAuth-iOS.git', :branch => 'master'`

### Allow Custom Browsers to be Opened
In the `info.plist` add:
```
<key>LSApplicationQueriesSchemes</key>
	<array>
		<string>googlechromes</string>
		<string>opera-https</string>
		<string>firefox</string>
	</array>
```

### Setup the App Identifier for the Browser to Redirect Back To
In the `info.plist` update to:
```
 			<key>CFBundleURLSchemes</key>
 			<array>
				<string>com.redhat.secure-ios-app.sso2</string>
 			</array>
```
Note, this should match what is defined as the app Id. (In Xcode click the project name in the top left, and the Bundle Identifier can be seen)

### Updating the AuthenticationService logic
Update the redirect URL:
`let REDIRECT_URL = URL(string:"com.redhat.secure-ios-app.sso2:/callback")` [Code](https://github.com/feedhenry/mobile-security-ios-template/blob/cbfac33503ce688c085b1064995ca67bf8158a39/secure-ios-app/services/AuthenticationService.swift#L24)

Update the authCall to include using the Safari browser or other: [Code](https://github.com/TommyJ1994/mobile-security-ios-template/blob/7ff1560b655fc6dceff28d92e5fdfca15802c207/secure-ios-app/services/AuthenticationService.swift#L51-L54)

```
let externalUserAgent = OIDExternalUserAgentIOSCustomBrowser.customBrowserSafari();

// this will automatically exchange the token to get the user info
self.currentAuthorisationFlow = OIDAuthState.authState(byPresenting: oidAuthRequest, externalUserAgent: externalUserAgent) {
...
```

1. Run the app and check auth works. 
2. Update all the URL's above to use `com.redhat.secure-ios-app.sso` to mimic a second app. Ensure that you also change the app ID for the project too. (In Xcode click the project name in the top left, and the Bundle Identifier can be seen). Then try the normal SSO flow and it should work.

I have added the code changes in my [master](https://github.com/TommyJ1994/mobile-security-ios-template) branch.

**Comment by wei-lee** [comment]
@TommyJ1994 so will this actually use the Safari browser then?

**Comment by tomjackman** [comment]
@wei-lee Yes, works fine with Safari. You can use other browsers like so:

```
let chromeBrowser = OIDExternalUserAgentIOSCustomBrowser.customBrowserChrome();
let firefoxBrowser = OIDExternalUserAgentIOSCustomBrowser.customBrowserFirefox();
let operaBrowser = OIDExternalUserAgentIOSCustomBrowser.customBrowserOpera();
```
